### PR TITLE
General maintenance

### PR DESCRIPTION
**Pull request by ocboogie**
#### Proposed
  * Fix non default imports. For example:
  ```js
  import path from 'path'
  // path has no default export so we would want to use
  import * as path from 'path'
  ```
  * Change comma-dangle in eslint
  * Change `lint-fix` npm script name to `format`
#### Todo
  * Move main and renderer source files to different folders
  * Move webpack configs to internals with prod and dev subfolder
  * Remove all `webpack.EnvironmentPlugin({NODE_ENV: ...})` webpack 4 takes care of that
  * Improve environment variable consistency
#### Changed
  * Move all build files to the dist folder
  * Add a `.prettierrc` and `.prettierignore` for better editor support
  * Integrate prettier into eslint
  * Fix Import order
  * Change all `path.resolve` to `path.join` for more consistency
  * Add html and json eslint plugins
  

Old pull request [here](https://github.com/chentsulin/electron-react-boilerplate/pull/1225).

Updated clean up @chentsulin.

Don't merge yet still need to change some things, submitted this to get feedback.

### CODE REVIEW

**Comment by ocboogie** [comment]
Does anyone know why this is failing?
I can't reproduce it

**Comment by ocboogie** [comment]
Sorry for the confusion? Better solution to what? @amilajack

**Comment by ocboogie** [comment]
It also might be a good idea to add prettier html and json to the git hooks

**Comment by ocboogie** [comment]
I can add those plugins as well if you would like

**Comment by amilajack** [comment]
Yes, please add them and autofix the styles

**Comment by ocboogie** [comment]
@amilajack Done

File: app/app.html [hunk 1]
```diff
@@ -5,15 +5,16 @@
   <meta charset="utf-8">
   <title>Hello Electron React!</title>
   <script>
-    (function () {
+    // eslint-disable-next-line func-names
+    (function() {
```
Comment (amilajack) [inline app/app.html]: Change this to:
```js
(() => {
  if (!process.env.HOT) {
    const link = document.createElement('link');
    link.rel = 'stylesheet';
    link.href = './dist/style.css';
    // HACK: Writing the script path should be done with webpack
    document.getElementsByTagName('head')[0].appendChild(link);
  }
})();
```
and remove the `eslint-disable` annotation

**Comment by amilajack** [comment]
Also please fix the merge conflicts

**Comment by ocboogie** [comment]
I fix the merge conflicts but I need to fix the problem with `postinstall`, specifically `electron-builder install-app-deps`

**Comment by amilajack** [comment]
Closing this for now until merge conflicts are fixed

**Pull request closed without merging**